spec/models/user_spec.rb

```diff
@@ -0,0 +1,110 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  let(:user_mock) { build(:user) }
+  let!(:user_mock2) { create(:user) }
+  let(:updated_user) do
+    create(
+      :user,
+      email: user_mock.email,
+      password: user_mock.password
+    )
+  end
+
+  context 'creation' do
+    it 'will mark an empty user instance as invalid' do
+      user = User.new
+      user.valid?
+      expect(user.errors.details[:email]).to include(
+        a_hash_including(error: :blank)
+      )
+    end
+
+    it 'will mark entries without an email as invalid' do
+      user = User.new(password: user_mock.password)
+      user.valid?
+      expect(user.errors.details[:email]).to include(
+        a_hash_including(error: :blank)
+      )
+    end
+
+    it 'will mark an empty password as invalid' do
+      user = User.new(email: user_mock.email)
+      user.valid?
+      expect(user.errors.details[:password]).to include(
+        a_hash_including(error: :blank)
+      )
+    end
+
+    it 'will mark a user with valid credentials as valid' do
+      user = User.new(
+        email: user_mock.email,
+        password: user_mock.password
+      )
+      expect(user.valid?).to eq(true)
```

Comment:
```suggestion
      expect(user).to be_valid
```

We have a shorthand for this :slightly_smiling_face: